extractor/TextExtractor_onepage.py

```diff
@@ -0,0 +1,48 @@
+from abc import ABC, abstractmethod
+from typing import List
+from bs4 import BeautifulSoup
+import requests
+import textwrap
+
+class Extractor(ABC):
+    @abstractmethod
+    def extract(self):
+        pass
+
+class Text:
+    def __init__(self, source: str, content: str):
+        self.source = source
+        self.content = content
+
+class WikipediaExtractor(Extractor):
+    def __init__(self, url: str):
+        self.url = url
+
+    def extract(self) -> List[Text]:
+        return self._extract_page(self.url)
+
+    def _extract_page(self, url: str) -> List[Text]:
+        response = requests.get(url)
+        soup = BeautifulSoup(response.text, 'html.parser')
+
+        # Entfernt alle sup-Tags, die Einzelnachweise enthalten (Zahlen hinter Text)
+        for sup in soup.find_all('sup'):
+            sup.decompose()
+
+        content_div = soup.find('div', id='mw-content-text')
+        if content_div is not None:
+            content = ' '.join(p.text for p in content_div.find_all('p'))
+        else:
+            content = ''
+
+        return [Text(url, content)]
+
```

Comment:
Da demnächst dein Modul jetzt auch von anderen Modulen verwendet wird, eine kleine Einführung in die `main`-Klausel:

Manchmal siehst du eine Zeile mit `if __name__ == "__main__":`.
Ohne dich in technische Details zu verwickeln, hier was diese Zeile tut und warum sie notwendig ist:

Durch die Zeile wird der Code innerhalb der If-Clause __nur__ ausgeführt, wenn dieses Modul __selbst__ ausgeführt wird, aber nicht, wenn es nur von anderen modulen importiert wird.

Wenn die Klausel nicht da ist, wird jeglicher Code der hier steht, bei jedem import dieses Moduls ausgeführt, was dann in Zukunft bedeuten würde, dass, egal was man eigentlich erreichen will, bei jedem Zugriff auf dieses Modul von außen, erstmal der Python Wikipedia Artikel extrahiert wird.


Jetzt musst du entsprechend deinen ganzen Code, der nur zur Demonstration dieses Moduls da ist, genau in so eine Klausel stecken.